nupic/research/frameworks/dendrites/routing/learn.py

```diff
@@ -259,18 +321,48 @@ def learn_to_route(
         print("{},{}".format(
             epoch, results["mean_abs_err"]
         ))
+        # track loss for plotting
+        if plot:
+            losses.append(results["mean_abs_err"])
+        # save models for future visualization or training
+        if save_interval and epoch % save_interval == 0:
+            if not os.path.exists(save_path):
+                os.makedirs(save_path)
+            torch.save(dendrite_layer.state_dict(),
+                       save_path + "dendrite_" + str(epoch))
+            if context_model:
+                torch.save(context_model.state_dict(),
+                           save_path + "context_" + str(epoch))
+
+    if plot:
+        import matplotlib.pyplot as plt
+        losses = np.array(losses)
+        plt.scatter(x=np.arange(1, num_training_epochs + 1), y=losses)
+        plt.savefig("training_curve.png")
+        plt.show()
```

Comment:
Why call `plt.show()`? You already saved the image, so I think this last line can be omitted. Also, in case this script isn't being run locally, I don't think this will have any effect.

What you can do is make showing the image optional by adding an additional parameter (which, by default, doesn't show the image).